src/NeoSharp.Core/Messaging/IMessageHandler[TMessage].cs

```diff
@@ -3,8 +3,16 @@
 
 namespace NeoSharp.Core.Messaging
 {
-    public interface IMessageHandler<in TMessage> where TMessage : Message
+    public interface IMessageHandler<in TMessage> : IMessageHandler
+        where TMessage : Message
     {
         Task Handle(TMessage message, IPeer sender);
     }
+
+    public interface IMessageHandler
```

Comment:
It is not type safe version.

Reply:
What you mean "not type safe"? we  don't need the message type.

Reply:
I mean the handler can accept a message with any Message type. We do need message type because the first thing we do inside Handle method is casting to the right type

Reply:
for sure we can find a solution for that .. using the IContainer inside this class was not right where there is patterns that solve this without using containers.

Would be nice to discuss a better solution about this where we don't use the IContainer, where er don't create the class already with the constructor and where we don't use know patterns like Strategy.

Reply:
I am not a fan to pass IContainer as a dependency and it is the very last resort for me but, as you can see, we either do it and benefit from strongly typed code or we don't and have weakly typed code. My preference is the former. That being said, I would recommend reverting this change (untyped IMessageHandler interface) until we have a full identical solution. I prefer not to replace the working code with partial one backed by the issue.

Reply:
Just to tell you the working code is not working .. that why was the change!!!

I'm glad that you aren't a fan of patterns ... then we can discuss that.
This PR is to fix something that wasn't working and wasn't against the code guidelines promoted by @gubanotorious indicating that we need to use known patterns as much we can.